src/Rootfs.jl

```diff
@@ -1100,3 +1107,83 @@ function installed_shards()
 
     return all_compiler_shards()[idx]
 end
+
+"""
+    manage_shards(; sort_by=:name, rev=false)
+
+Open a prompt allowing a user to selectively remove downloaded compiler shards.
+By default, the shards are sorted by name, alternatively you can sort them by
+file size on disk by specifying `sort_by=:size`. With `rev=true` you can
+reverse the sort order.
+"""
+function manage_shards(; sort_by=:name, rev=false)
+    # Get all installed shards
+    shards = installed_shards()
+
+    # Obtain directory sizes for the artifacts
+    totalsizes = totalsize.(Artifacts.artifact_path.(shard_source_artifact_hash.(shards)))
+
+    # Sort shards and totalsizes
+    if sort_by === :name
+        perm = sortperm(shards, rev=rev)
+    elseif sort_by == :size
+        perm = sortperm(totalsizes, rev=rev)
+    else
+      error("unsupported sort value: :$sort_by (allowed: :name, :size)")
+    end
+    shards .= shards[perm]
+    totalsizes .= totalsizes[perm]
+
+    # Build menu items
+    menuitems = similar(shards, String)
+    for i in 1:length(shards)
+        menuitems[i] = @sprintf("(%5.3f GiB) %s", totalsizes[i] / 2^30, shards[i])
+    end
+
+    # Prompt user
+    ts = @sprintf("%5.3f", sum(totalsizes) / 2^30)
+    manage_shards_menu = TerminalMenus.request(
+        "Which compiler shards should be removed (total size on disk: $ts GiB)?",
+        TerminalMenus.MultiSelectMenu(menuitems, pagesize = 10; charset=:ascii)
+    )
+
+    # Handle no selection
+    if isempty(manage_shards_menu)
+        println("Removed 0 compiler shards.")
+        return nothing
+    end
+
+    # Otherwise prompt for confirmation
+    println("\nThe following shards have been marked for removal:\n")
+    for item in menuitems[sort(Int.(manage_shards_menu))]
+        println("  ", item)
+    end
+    print("\nAre you sure that these should be removed? [Y/n]: ")
+    answer = lowercase(strip(readline()))
+
+    # If removal is confirmed, deleting the relevant artifacts
+    if isempty(answer) || answer == "yes" || answer == "y"
+        Pkg.remove_artifact.(shard_source_artifact_hash.(shards[Int.(manage_shards_menu)]))
+        println("Removed ", length(manage_shards_menu), " compiler shards.")
+    else
+        println("Removed 0 compiler shards.")
+    end
+end
+
+# Return total size on a disk of a file or directory
+function totalsize(path)
+  if isfile(path)
```

Comment:
Uhm, `isfile` isn't complementary to `isdir`, for example
```julia
julia> isfile("/dev/stdout")
false

julia> isdir("/dev/stdout")
false
```
If I understand correctly here you're assuming that if something `!isfile` then `isdir`?

Reply:
You're right. I've changed it to `!isdir` in 1f8b1b4, which seems to be closer semantically to what I want (i.e., do not recurse if it does not identify as a directory).